projects/exui-common-lib/src/lib/services/google-tag-manager/google-tag-manager.service.ts

```diff
@@ -60,10 +58,14 @@ export class GoogleTagManagerService {
     }
   }
 
-  public event(eventName: string, params: {}) {
-    (this.window as any).dataLayer.push({
-      event: eventName,
-      params
-    });
+  public event(eventName: string, params: Record<string, any>) {
+    if ((this.window as any).dataLayer) {
+      (this.window as any).dataLayer.push({
+        event: eventName,
+        ...params
+      });
+    } else {
+      console.warn('Google Analytics not initialized: dataLayer is undefined');
```

Comment:
We'll need to be careful this doesn't end up spamming the console. OK for now but probably remove before release